src/components/Carousel.tsx

```diff
@@ -1,18 +1,101 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinity: boolean,
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinity,
+}) => {
+  const [offset, setOffset] = useState(0);
+
+  const handlePrevClick = (prev: number) => {
+    if (prev >= 0 && infinity) {
+      return -(itemWidth * images.length) + itemWidth * frameSize;
+    }
+
+    if (prev < -(itemWidth * step)) {
+      return prev + step * itemWidth;
+    }
+
+    return 0;
+  };
+
+  const handleNextClick = (prev: number) => {
+    if (prev <= -(images.length * itemWidth) + itemWidth * frameSize
+      && infinity
+    ) {
+      return 0;
+    }
+
+    if (prev <= -(images.length * itemWidth) + itemWidth * (frameSize + step)) {
+      return -(images.length * itemWidth) + itemWidth * frameSize;
+    }
+
+    if (prev >= -(images.length * itemWidth) + itemWidth * frameSize) {
+      return prev - itemWidth * step;
+    }
+
+    return 0;
+  };
+
+  return (
+    <div
+      className="Carousel"
+      style={{
+        width: itemWidth * frameSize,
+      }}
+    >
+      <ul
+        className="Carousel__list"
+        style={{
+          transform: `translateX(${offset}px)`,
+          transition: `transform ${animationDuration}ms`,
+        }}
+      >
+        {images.map(image => (
+          <li
+            key={image}
+          >
+            <img
+              src={image}
+              alt="img"
+              width={itemWidth}
+            />
+          </li>
+        ))}
+      </ul>
+
+      <button
+        type="button"
+        onClick={() => (
+          setOffset(handlePrevClick)
+        )}
```

Comment:
consider to make one handler factory which will except callback and return arrow function.
```
const handleOffset = (callback: React.Dispatch<React.SetStateAction<number>>) => () => {
  setOffset(callback)
}
```
```suggestion
        onClick={handleOffset(handlePrevClick)}
```